### PR TITLE
feat(components): image-viewer component support slot

### LINKED ISSUE #3106: [Feature Request] Can you provide a scope slot so that we can add custom content to the image viewer?
### Existing Component
Yes

### Component Name
el-image

### Description
Sometimes, we want to display some text information in the current preview image during image preview, but the API currently provided cannot do so. Therefore, can you provide a scope slot so that we can add custom content to the image viewer?

### PR DESCRIPTION
**Pull request by xiaoxian521**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by xiaoxian521** [comment]
feat: https://github.com/element-plus/element-plus/issues/3106

File: packages/components/image/src/index.vue [hunk 1]
```diff
@@ -32,7 +32,9 @@
           :hide-on-click-modal="hideOnClickModal"
           @close="closeViewer"
           @switch="switchViewer"
-        />
+        >
+          <slot />
```
Comment (sxzz) [inline packages/components/image/src/index.vue:36]: maybe we should use a named slot, such as `viewer`. WDYT?

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
LGTM. Could you add it to docs later?

**Comment by xiaoxian521** [comment]
> LGTM. Could you add it to docs later?

ok

**Comment by JeremyWuuuuu** [comment]
> > LGTM. Could you add it to docs later?
> 
> ok

Please raise an issue for this in case this is forgotten.

**Pull request merged**